go/master/c/client.go

```diff
@@ -84,11 +85,29 @@ func paddle_set_dataset(client C.paddle_master_client, path **C.char, size C.int
 	return C.PADDLE_MASTER_OK
 }
 
+// return value:
+//     0:ok
+//    -1:EOF
+//    -2:error
 //export paddle_next_record
 func paddle_next_record(client C.paddle_master_client, record **C.uchar) C.int {
 	c := get(client)
-	r := c.NextRecord()
+	r, err := c.NextRecord()
+	if err == io.EOF {
+		// EOF
+		*record = (*C.uchar)(nullPtr)
+		return -1
+	}
+
+	if err != nil {
+		// Error
+		// TODO: return the type of error?
```

Comment:
I don't think C side can do much if we return the kind of error. How about we remove the TODO, and do `log.Errorln(err)` here?